bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/TableViewer.java

```diff
@@ -459,41 +459,51 @@ protected Widget doFindItem(Object element) {
 
 	@Override
 	void handleExpandableNodeClicked(Widget w) {
-		if (!(w instanceof Item item)) {
-			return;
-		}
-
-		Object data = item.getData();
+		boolean oldBusy = isBusy();
+		try {
+			setBusy(true);
+			this.getTable().setRedraw(false);
```

Comment:
As discussed, `setRedraw(false)` is not needed (here and in the TreeViewer), so please remove from patch.

Reply:
**Without setRedraw(false)**
time taken to expand :242
items created :20001
time taken to expand :395
items created :20001
time taken to expand :269
items created :20001
time taken to expand :328
items created :20001
time taken to expand :396
items created :20001
time taken to expand :467
items created :20001
time taken to expand :767
items created :20001
time taken to expand :603
items created :20001
time taken to expand :672
items created :20000

**with setRedraw(false)**
time taken to expand :105
items created :20001
time taken to expand :103
items created :20001
time taken to expand :101
items created :20001
time taken to expand :108
items created :20001
time taken to expand :100
items created :20001
time taken to expand :99
items created :20001
time taken to expand :99
items created :20001
time taken to expand :98
items created :20001
time taken to expand :97
items created :20000

So this data proves that `setRedraw(false)` works for Table

Reply:
Thanks!